calico_node/startup/startup.go

```diff
@@ -762,17 +813,55 @@ func ensureFilesystemAsExpected() {
 
 // ensureDefaultConfig ensures all of the required default settings are
 // configured.
-func ensureDefaultConfig(cfg *api.CalicoAPIConfig, c *client.Client, node *api.Node) error {
+func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface, node *api.Node) error {
 	// By default we set the global reporting interval to 0 - this is
 	// different from the defaults defined in Felix.
 	//
 	// Logging to file is disabled in the felix.cfg config file.  This
 	// should always be disabled for calico/node.  By default we log to
 	// screen - set the default logging value that we desire.
-	if err := ensureGlobalFelixConfig(c, "ReportingIntervalSecs", "0"); err != nil {
-		return err
-	} else if err = ensureGlobalFelixConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
-		return err
+	felixConf, err := c.FelixConfigurations().Get(ctx, globalFelixConfigName, options.GetOptions{})
```

Comment:
I think we should revisit this code later.  When code is this nested it usually means it can be refactored to improve readability.  It is quite difficult to work through.

Let's leave for now, but track a tech debt item to refactor some of this - possibly by introducing some patch mechanism or something.  Not sure.